great_expectations/execution_engine/split_and_sample/data_sampler.py

```diff
@@ -75,8 +79,8 @@ def verify_batch_spec_sampling_kwargs_key_exists(
     def get_sampling_kwargs_value_or_default(
         batch_spec: BatchSpec,
         sampling_kwargs_key: str,
-        default_value: Optional[Any] = None,
-    ) -> Optional[Any]:
+        default_value: Optional[T] = None,
+    ) -> T | Any:
```

Comment:
Should this be returning `T` or `Any`? Is `T` not a member of `Any`?

Reply:
`T` is only representing the `default_value`, if it is provided.